bundle/src/main/webapp/app-root/clientlibs/io.wcm.caconfig.editor/js/widgets/tagbrowser.directive.js

```diff
@@ -0,0 +1,248 @@
+/*
+ * #%L
+ * wcm.io
+ * %%
+ * Copyright (C) 2022 wcm.io
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+(function (angular, Coral, $) {
+  "use strict";
+
+  angular.module("io.wcm.caconfig.widgets")
+    .directive("caconfigTagbrowser", tagbrowser);
+
+  tagbrowser.$inject = ["templateUrlList", "directivePropertyPrefixes", "$timeout", "$rootScope", "configService", "tagbrowserService"];
+
+  function tagbrowser(templateList, directivePropertyPrefixes, $timeout, $rootScope, configService, tagbrowserService) {
+    var directive = {
+      replace: true,
+      templateUrl: templateList.tagbrowser,
+      scope: {
+        property: "="
+      },
+      link: link
+    };
+
+    return directive;
+
+    function link(scope, element) {
+      var multiValue = scope.property.metadata.multivalue;
+
+      if(multiValue) {
+        scope.effectiveValues = [];
+        scope.values = [];
+
+        setValueArray(scope.property.effectiveValue, scope.effectiveValues);
+        setValueArray(scope.property.value, scope.values);
+      }
+
+      var prefix = directivePropertyPrefixes.tagbrowser;
+      var props = scope.property.metadata.properties;
+
+      var tagfieldName = "tags-" + Math.floor(Math.random() * 100000);
```

Comment:
huh, i think there is a better solution than this. probably we can use something like an UUID here?

Reply:
There is perfect way to do this in JS, afaik.  You could introduce a new module to do this but that's a lot of overhead.  
The `Math.floor + Math.random` way is a proven way to do this in JS: https://dev.to/rahmanfadhil/how-to-generate-unique-id-in-javascript-1b13].

Fast, reliable and little overhead.

If you feel this really is something you want changed, I will do it.  Please let me know your preferred way then.

Reply:
probably `Coral.commons.getUID()` does the job?

Reply:
It gives an id like `tags-coral-id-99`.  If that's OK with you, I'll commit it.
It's definitely more elegant.

Reply:
yes, let's use it. it's used throughout the AEM code.